### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by saketkum11**
Please Review Cart Mangement.

- addCart
- removeCart
- PriceCalculation
- cart context
- catergories
- move to cart
- move to wishlist

### CODE REVIEW

File: src/Pages/Home/Home.jsx [hunk 1]
```diff
@@ -1,74 +1,101 @@
...
-            <Link to="/product"
-                className="text-color-9 text-dec text-m rounded-m pd-x-9 pd-y-3 position-ab bg-black-1 p-t-7 p-lf-1">
+  const [categoriesValue, setCategoriesValue] = useState([]);
+
+  useEffect(() => {
+    const categories = async () => {
+      try {
+        const response = await axios.get('/api/categories');
+        setCategoriesValue(response.data.categories);
+        console.log(response);
```
Comment (JowelTisso) [inline src/Pages/Home/Home.jsx]: Remember to remove console logs

**Review by JowelTisso — APPROVED** [review]
* You can extract axios functions separately
* Rest Looks Good

File: src/Pages/Home/Home.jsx (hunk 1 above)
Comment (saketkum11) [reply src/Pages/Home/Home.jsx]: ok sure.

File: src/Component/Card/Card.jsx [hunk 2]
```diff
@@ -1,13 +1,14 @@
 import React from 'react';
+import { useCart } from '../../Context/Cart/Cart-Context';
 import { useFilter } from '../../Context/Filter/Filter-Context';
 import { useProduct } from '../../Context/Product/Product-Context';
-
 import { useWishlist } from '../../Context/Wishlist/Wishlist-Context';
```
Comment (narharikale) [inline src/Component/Card/Card.jsx:5]: you can use barrel to export for these imports

File: src/Component/Card/Card.jsx [hunk 3]
```diff
@@ -59,27 +60,46 @@ function Card() {
...
                         <small className=' text-sm'>outofStock</small>
                       )}
-                      <small className=' text-sm'>{quantity}</small>
+
+                      <small className=' text-s'>Rs.{price}</small>
+                      <small className=' text-s'>{quantity}</small>
+
                       <small className=' text-sm'>{categoryName}</small>
-                      <small className=' text-sm'>rating - {rating}/5</small>
+                      <div class='bg-green-7 pd-2 m-y-5'>
```
Comment (narharikale) [inline src/Component/Card/Card.jsx:77]: i think you missed , class shoul be className

File: src/Component/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -46,18 +50,22 @@ function Navbar() {
                 <Link to='/wishlist' className='text-dec text-color-0 cursor'>
                   <i className='fa-solid  fa-heart text-color-0'></i>
                 </Link>
-                <span className=' text-s position-ab p-t-n-2 flex items-center justify-center bg-red-7 wt-fixed-5 h-fixed-5 rounded-full p-r-0'>
-                  0
-                </span>
+                {wishlist.length >= 0 && (
+                  <span className=' text-s position-ab p-t-n-2 flex items-center justify-center bg-red-7 wt-fixed-5 h-fixed-5 rounded-full p-r-0'>
```
Comment (narharikale) [inline src/Component/Navbar/Navbar.jsx:54]: we can make separate class for this, as we are more properties

File: src/Pages/Cart/Cart.jsx [hunk 5]
```diff
@@ -1,91 +1,141 @@
...
+                  imageSrc,
+                  title,
+                  price,
+                  inStock,
+                  quantity,
+                  rating,
+                  qty,
+                  discount,
+                  categoryName,
+                } = product;
```
Comment (narharikale) [inline src/Pages/Cart/Cart.jsx:37]: i think you need to configure your fomatter extension

**Review by narharikale — APPROVED** [review]
good work , 👍

File: src/Component/Card/Card.jsx (hunk 2 above)
Comment (saketkum11) [reply src/Component/Card/Card.jsx:5]: okkk sure will do this

File: src/Component/Card/Card.jsx (hunk 3 above)
Comment (saketkum11) [reply src/Component/Card/Card.jsx:77]: Yes , I will do that .

File: src/Pages/Cart/Cart.jsx (hunk 5 above)
Comment (saketkum11) [reply src/Pages/Cart/Cart.jsx:37]: prettier is working thats why is like that.

**Pull request merged**